File: 087/b.py
# B - Coins
# https://atcoder.jp/contests/abc087/tasks/abc087_b

# A 500円玉
# B 100円 
# C 50円
A = int(input())
B = int(input())
C = int(input())
X = int(input())

i = 0

for a in range(A+1):
    if (a * 500) == X:
        i += 1
        break

    for b in range(B+1):
        if (a * 500) + (b * 100) == X:
            i += 1
            break

        for c in range(C+1):
            if (a * 500) + (b * 100) + (c * 50)== X:
                i += 1
                break

print(i)

File: 087/b.rb
# B - Coins
# https://atcoder.jp/contests/abc087/tasks/abc087_b

x = gets.strip.to_i
y = gets.strip.to_i
z = gets.strip.to_i
t = gets.strip.to_i

# a, b, c, x = 4.times.map { gets.to_i } <= スマートなやり方

cnt = 0

(0..x).each do |i|
    (0..y).each do |j|
        (0..z).each do |k|
            if (i*500 + j*100 + k*50) == t
                cnt = cnt +1
            end
        end
    end
end

p cnt